modules/barcode/tutorials/barcode_detect_and_decode/barcode_detect_and_decode.markdown

```diff
@@ -0,0 +1,129 @@
+Bar code Recognition{#tutorial_barcode_detect_and_decode}
+======================
+
+Goal
+----
+
+In this chapter,
+
+-   We will familiarize with the bar code detection and decoding methods available in OpenCV.
+
+Basics
+----
+
+Bar code is major technique to identify commodity in real life.  A common bar code is a pattern of parallel lines arranged by black bars and white bars with vastly different reflectivity. Bar code recognition is to scan the bar code in the horizontal direction to get a string of binary codes composed of bars of different widths and colors, that is, the code information of the bar code. The content of bar code can be decoded by matching with various bar code encoding methods. For current work, we only support EAN13 encoding method.
+
+### EAN 13
+
+The EAN-13 bar code is based on the UPC-A standard, which was first implemented in Europe by the International Item Coding Association and later gradually spread worldwide. Most of the common goods in life use EAN-13 barcode.
+
+for more detail see [EAN - Wikipedia](https://en.wikipedia.org/wiki/International_Article_Number)
+
+### BarcodeDetector
+Several algorithms were introduced for bar code recognition.
+
+While coding, we firstly need to create a **cv::barcode::BarcodeDetector** object.  It has mainly three member functions, which will be introduced in the following.
+
+#### detect
+
+It is a algorithm based on directional consistency. First of all, we compute the average squared gradients of every pixels. It was proposed in the paper "Systematic methods for the computation of the directional  fields and singular points of fingerprints" by A.M. Bazen and S.H. Gerez in 2002. Then we divide the image into some square patches and compute the **gradient variance** and **mean gradient direction** of each patch. At last we connected the patches that have **low gradient variance** and **similar gradient direction**. In this stage, we use multi-size patches to capture the gradient distribution of multi-size bar codes, and apply non-maximum suppression to filter duplicate proposals.
+
+See a simple example below:
+
+@code{.cpp}
+#include "opencv2/barcode.hpp"
+#include "opencv2/imgproc.hpp"
+
+using namespace cv;
+barcode::BarcodeDetector bardet;
+Mat frame = imread("bar_code.jpg");
+std::vector<Point> corners;
+bool ok = bardet.detect(frame, corners);
+@endcode
```

Comment:
Please do not put code into the documentation directly. This code may not be updated on changes (it is not checked for compilation).

Use `@snippet` / `@include` doxygen statements for embedding code from .cpp files (see corresponding examples in other OpenCV modules).